realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -582,7 +582,8 @@ public double average(String fieldName) {
         if (tableOrView instanceof Table) {
             tableView = ((Table) tableOrView).getDistinctView(columnIndex);
         } else {
-            tableView = ((TableView) tableOrView).getTable().getDistinctView(columnIndex);
+            tableView = ((TableView) tableOrView).getCopy();
```

Comment:
I am really surprised that we need to copy a tableview to apply multiple distinct parameters. It sounds like a bug in Core to me or at least a future enhancement. We should verify what the intended behaviour of multiple distinct calls is.